app/routes.py

```diff
@@ -1,2 +1,198 @@
+    db.session.add(new_task)
+    db.session.commit()
+
+    return jsonify({"task": new_task.to_dict()}), 201
+
+@tasks_bp.route("", methods=["GET"])
+def handle_tasks():
+    tasks_response = []
+    if request.args.get("sort")=="asc":
+        tasks=Task.query.order_by(Task.title)
+    elif request.args.get("sort")=="desc":
+        tasks=Task.query.order_by(Task.title.desc())
+   
+    else:
+        tasks = Task.query.all()
+    for task in tasks:
+        tasks_response.append(task.to_dict())
+    return jsonify(tasks_response), 200
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT"])
+def handle_task(task_id):
+    task_id=validate_id_int(task_id)
+    # task_id = int(task_id)
+    task = Task.query.get(task_id)
+    if not task:
+        return make_response("", 404)
+    if request.method=="GET":
+        return jsonify({"task": task.to_dict()}), 200
+    elif request.method== "PUT":
+        request_data=request.get_json()
+        task.title=request_data["title"]
+        task.description=request_data["description"]
+    
+        db.session.commit()
+        return jsonify({"task": task.to_dict()}),200
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def complete_task(task_id):
+    task_id=validate_id_int(task_id)
+    task = Task.query.get(task_id)
+    patch_data=request.get_json()
+    # print(patch_data)
+    if not task:
+        return make_response("", 404)
+    
+    else:
+        task.completed_at=date.today()
+
+        db.session.commit()
+        
+        my_token=os.environ.get("MY_TOKEN")
+        message=f"Someone just completed the task {task.title}"
+        query={"channel": "task-notifications", "text": message}
+        my_headers={"Authorization": f"Bearer {my_token}"}
+        requests.post("https://slack.com/api/chat.postMessage", data=query, headers= my_headers)
+    
+        
+        return jsonify({"task": task.to_dict()}),200
+ 
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def incomplete_task(task_id):
+    task_id=validate_id_int(task_id)
+    task = Task.query.get(task_id)
+    patch_data=request.get_json()
+    # print(patch_data)
+    if not task:
+        return make_response("", 404)
+    
+    else:
+        task.completed_at= None
+
+        db.session.commit()
+        return jsonify({"task": task.to_dict()}),200
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    task_id=validate_id_int(task_id)
+    
+    task = Task.query.get(task_id)
+
+    if task:
+        db.session.delete(task)
+        db.session.commit()
+        return {"details": f'Task {task_id} "{task.title}" successfully deleted'}, 200
+    else:
+        abort (404)
+
+def validate_id_int(task_id):
+    try:
+        task_id = int(task_id)
+        return (task_id)
+    except:
+        abort(400, "Error: Task ID needs to be a number")
+
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
```

Comment:
It might be a good idea to put the `goals` and `tasks` routes into separate files, as this is getting pretty long.